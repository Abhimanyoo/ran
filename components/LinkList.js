import Link from 'next/link';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const LinkList = ({ className, pathname }) =>
  <nav className={className}>
    <Link prefetch href="/">
      <a className={pathname === '/' && 'is-active'}>Main Page</a>
    </Link>
    <Link prefetch href="/create">
      <a className={pathname === '/create' && 'is-active'}>Create</a>
    </Link>
    <a
      href="https://github.com/Sly777/ran"
      rel="noopener noreferrer"
      target="_blank"
    >
      RAN! @ Github
    </a>
  </nav>;

LinkList.propTypes = {
  pathname: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired
};

export default styled(LinkList)`
  a {
    font-size: 14px;
    margin-right: 15px;
    text-decoration: none;

    &.is-active {
      text-decoration: underline;
    }
  }
`;
