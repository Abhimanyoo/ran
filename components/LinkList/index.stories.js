import React from 'react';

import { storiesOf } from '@storybook/react';
import LinkList from '.';

storiesOf('Menu', module).add('LinkList', () => <LinkList />);
