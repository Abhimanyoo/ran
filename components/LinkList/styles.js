import { A, Button } from '~/components/Theme';

export const StyledA = A.extend`
  font-size: 14px;
  margin-right: 15px;
  text-decoration: none;
  cursor: pointer;
  text-decoration: ${({ active }) => (active ? 'underline' : 'none')};
`;

export const LogOutButton = Button.extend`
  display: inline-block;
  margin-right: 15px;
  cursor: pointer;
  text-decoration: ${({ active }) => (active ? 'underline' : 'none')};
`;
