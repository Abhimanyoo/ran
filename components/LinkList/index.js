import PropTypes from 'prop-types';
import { Link } from '../../routes';
import { StyledA, LogOutButton } from './styles';

const LinkList = ({ pathname, authenticated, logout }) =>
  <nav>
    <Link prefetch href="/" passHref>
      <StyledA active={pathname === '/'}>Main Page</StyledA>
    </Link>
    <Link prefetch route="create" passHref>
      <StyledA active={pathname === '/create_post'}>Create</StyledA>
    </Link>
    {!authenticated &&
      <Link prefetch route="signin" passHref>
        <StyledA active={pathname === '/sign_in'}>SignIn</StyledA>
      </Link>}
    {!authenticated &&
      <Link prefetch route="signup" passHref>
        <StyledA active={pathname === '/sign_up'}>SignUp</StyledA>
      </Link>}
    {authenticated &&
      <LogOutButton
        role="link"
        href="#"
        onClick={() => logout()}
        active={pathname === '/sign_up'}
      >
        LogOut
      </LogOutButton>}
    <StyledA
      href="https://github.com/Sly777/ran"
      rel="noopener noreferrer"
      target="_blank"
    >
      RAN! @ Github
    </StyledA>
  </nav>;

LinkList.propTypes = {
  pathname: PropTypes.string.isRequired,
  authenticated: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired
};

export default LinkList;
