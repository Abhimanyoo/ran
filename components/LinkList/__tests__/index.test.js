import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import LinkList from '../';
import { defaultProps } from './__mocks__/index.mock';

describe('<LinkList />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<LinkList {...defaultProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('when not authenticated', () => {
    const wrapper = shallow(
      <LinkList {...defaultProps} authenticated={false} />
    );
    const html = wrapper.html();

    expect(html).toEqual(expect.stringContaining('Main Page'));
    expect(html).toEqual(expect.stringContaining('Create'));
    expect(html).toEqual(expect.stringContaining('SignIn'));
    expect(html).toEqual(expect.stringContaining('SignUp'));
    expect(html).not.toEqual(expect.stringContaining('LogOut'));
  });

  it('when authenticated', () => {
    const wrapper = shallow(<LinkList {...defaultProps} authenticated />);
    const html = wrapper.html();

    expect(html).toEqual(expect.stringContaining('Main Page'));
    expect(html).toEqual(expect.stringContaining('Create'));
    expect(html).not.toEqual(expect.stringContaining('SignIn'));
    expect(html).not.toEqual(expect.stringContaining('SignUp'));
    expect(html).toEqual(expect.stringContaining('LogOut'));
  });
});
