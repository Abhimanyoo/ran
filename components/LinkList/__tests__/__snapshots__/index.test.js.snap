// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkList /> should render with default props 1`] = `
<nav>
  <LinkRoutes
    href="/"
    passHref={true}
    prefetch={true}
  >
    <styled.a
      active={false}
    >
      Main Page
    </styled.a>
  </LinkRoutes>
  <LinkRoutes
    passHref={true}
    prefetch={true}
    route="create"
  >
    <styled.a
      active={false}
    >
      Create
    </styled.a>
  </LinkRoutes>
  <LinkRoutes
    passHref={true}
    prefetch={true}
    route="signin"
  >
    <styled.a
      active={false}
    >
      SignIn
    </styled.a>
  </LinkRoutes>
  <LinkRoutes
    passHref={true}
    prefetch={true}
    route="signup"
  >
    <styled.a
      active={false}
    >
      SignUp
    </styled.a>
  </LinkRoutes>
  <a
    href="https://github.com/Sly777/ran"
    rel="noopener noreferrer"
    target="_blank"
  >
    RAN! @ Github
  </a>
</nav>
`;
